docs/source/en/api/loaders/single_file.md

```diff
@@ -10,28 +10,235 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
+
+```python
+from huggingface_hub import hf_hub_download, snapshot_download
+
+my_local_checkpoint_path = hf_hub_download(
+    repo_id="stabilityai/stable-diffusion-xl-base-1.0",
+    filename="sd_xl_base_1.0_0.9vae.safetensors"
+    local_dir="my_local_checkpoints",
+    local_dir_use_symlinks=False
+)
+print("My local checkpoint: ", my_local_checkpoint_path)
+
+my_local_config_path = snapshot_download(
+    repo_id="stabilityai/stable-diffusion-xl-base-1.0",
+    local_dir="my_local_sdxl_config",
+    local_dir_use_symlinks=False,
+    allowed_patterns=["*.json", "**/*.json", "*.txt", "**/*.txt"]
+)
+print("My local config: ", my_local_config_path)
 
-## FromOriginalVAEMixin
+```
 
-[[autodoc]] loaders.autoencoder.FromOriginalVAEMixin
+Then pass the local paths to the `pretrained_model_link_or_path` and `config` arguments of the `from_single_file` method.
+
+```python
+pipe = StableDiffusionXLPipeline.from_single_file(my_local_checkpoint_path, config=my_local_config_path, local_files_only=True)
+
+```
+
+<Tip>
+Disabling symlinking means that the `huggingface_hub` caching mechanism has no way to determine whether a file has already been downloaded to the local directory. This means that the `hf_hub_download` and `snapshot_download` functions will download the file to the local directory every time they are called. If you are disabling symlinking, it recommended that you separate the download and loading steps to avoid downloading the same file multiple times.
+
+</Tip>
+
+## Using the original configuration file of a model
+
+If you would like to use the original configuration file of a model when loading a model from a single file, you can do so with the `original_config` argument.
+
+```python
+from diffusers import StableDiffusionXLPipeline
+
+ckpt_path = "https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/sd_xl_base_1.0_0.9vae.safetensors"
+repo_id = "stabilityai/stable-diffusion-xl-base-1.0"
+original_config = "https://raw.githubusercontent.com/Stability-AI/generative-models/main/configs/inference/sd_xl_base.yaml"
+
+pipe = StableDiffusionXLPipeline.from_single_file(ckpt_path, original_config=original_config)
+```
+
+<Tip>
+When using `original_config` with local_files_only=True`, `diffusers` will attempt to infer the components based on the type signatures of pipeline class. This is not as reliable as providing a config path and might lead to errors when configuring the pipeline. Additionally, the pipeline scheduler will default to the `DDIMScheduler` if one isn't provided.
```

Comment:
If `original_config` is provided we will configure the models based on it, regardless of what `local_files_only` is set to.

The interaction between `local_files_only` and `original_config` helps us determine whether or not we can fetch the pipeline `model_index.json` files and individual model `config.json` files from the hub. 

Consider this existing functionality of `from_single_file`

```python
pipe = StableDiffusionPipeline.from_single_file(
    "../checkpoints/sd21/v2-1_768-ema-pruned.safetensors",
    original_config="../checkpoints/sd21/config.yaml",
    local_files_only=True
)
```

In this case, a local original config has been provided, along with a local model and `local_files_only=True`.  Assuming the CLIP model configs have been placed correctly in the cache folder, this snippet will load even if there is no internet connection. This is because we have currently hardcoded what components we load in `from_single_file` , which is limited to the components of the Stable Diffusion pipelines, so we know what we need to load.

We can consider this "legacy" local behaviour. In this situation that we assume we shouldn't attempt to fetch anything from the internet, or it might break existing code if there is no internet connection. Admittedly, this is probably an edge case, but it would be nicer to not try to break anything. This is why in the updated code in this PR, we make use of the pipeline signature keys. 

We could avoid this by always trying to load a model_index.json file regardless of what `local_files_only` is set to, but I think the expectation from a user is that we won't be connecting to the internet.